### DISCUSSION TITLE
SSO | eXo Login redirection to IDP

### DISCUSSION THREAD

**Discussion by Rajamuthukrishna** [exoplatform/exo-community#76] [Q&A]
Hello Team,

We have implemented openId SSO implementation with keyclock IDP. Initially eXo login page is showing, once click the sign in with single account the it will redirect to keyclock login page.

We have requirement for while use the eXo application login url, it should redirect to keyclock IDP instead of exo login page.
<img width="932" alt="image" src="https://github.com/exoplatform/exo-community/assets/112479162/275af26c-1c82-494c-bd8c-fc526713ebcd">

Regards,
Raja

**Comment by Rajamuthukrishna** [comment]
Hello Team,

We have implemented openId SSO Implemetation with keyclock IDP. 

while try to access the eXo url, eXo login page is showing, we have requirement of redirecting to keyclock IDP while click the eXo login.

kindly anyone help on this.
[exo.properties.txt](https://github.com/exoplatform/exo-community/files/15264976/exo.properties.txt)

**Comment by rdenarie** [comment]
Hello @Rajamuthukrishna 
With the OpenId implementation, it is not possible that the initial page is the IDP login page.
In the OpenId connect flow, you firstly arrive on exo login page, then click on the OpenId button, which redirect you to IDP login page.
A workaround could be : 
- Provide this url to your users : https://${hostname}/portal/openidAuth?_oauthInteraction=start&_initialURI=%2Fportal&_rememberme=true

This is the redirect url of the openid button on the exo login page. It will initiate the openid connection flow and send users directly to IDP login form


To access directly to the IDP login page, it is preferable to use SAML SSO implementation, which will automatically redirect on IDP login page without displaying the exo login page (but in this case the feature "External Users" will no be available if externals users are not in the IDP)


For the next time, can you ask you question on community.exoplatform.com ?
You have different spaces in which you can ask questions, and you will have more visibility.

**Reply by Rajamuthukrishna** [reply]
Thanks for the prompt response.
